server/server/views/event.py

```diff
@@ -48,18 +50,23 @@ def collection_post(self):
                 new_event.main_image = request.validated['main_image']
             if 'end_date' in request.validated:
                 new_event.end_date = request.validated['end_date']
-            category = Category.get_by_name(request,
-                                            request.validated['category'].
-                                            lower())
-            if category is not None:
-                new_event.category_id = category.id
-            else:
-                new_event.category_id = Category\
-                    .new_get_id(request, request.validated['category'].lower())
+            new_event.category_id = Category\
+                .get_by_name(request, request.validated['category'].lower()).id
+
             Event.add_event(request, obj=new_event)
             new_event_id = Event.get_event_obj(request,
                                                name=request.validated['name']
                                                ).id
+
+            if 'tags' in request.validated:
+                for tag in request.validated['tags']:
```

Comment:
`dict` has very useful method [get](https://docs.python.org/2/library/stdtypes.html#dict.get). So you may substitute 2 lines above by one:
```
for tag in request.validated.get('tags', []):
```